src/base/bittorrent/torrentinfo.cpp

```diff
@@ -374,7 +379,7 @@ QVector<QByteArray> TorrentInfo::pieceHashes() const
     hashes.reserve(count);
 
     for (int i = 0; i < count; ++i)
-        hashes += {m_nativeInfo->hash_for_piece_ptr(lt::piece_index_t {i}), InfoHash::length()};
+        hashes += {m_nativeInfo->hash_for_piece_ptr(lt::piece_index_t {i}), SHA1Hash::length()};
```

Comment:
I think you should add `TorrentID::length()`.

Reply:
IIRC, this isn't "torrent ID". It's just SHA1 hash of piece.
Anyway, `TorrentID::length()` is here (it's inherited).

Reply:
>IIRC, this isn't "torrent ID". It's just SHA1 hash of piece.

You're right.

But shouldn't libtorrent v2 return sha256 (in some other way)? it is weird that it still returns sha1...

Reply:
>But shouldn't libtorrent v2 return sha256 (in some other way)? it is weird that it still returns sha1...

I don't know. In addition, IIRC, the specification says about changing the hashing algorithm only for the info hash.

Reply:
I didn't go into it much. I just saw that libtorrent provides sha1 here without any conditions.

Reply:
>I suppose libtorrent didn't bother to update this function and they provided other ways to return the merkle tree.

May be. However, this function is not excluded from the new API.